### PR TITLE
Card part attributed text patch

### PR DESCRIPTION
**Pull request by jl-codes**
## Issue Link :link:
 <ul>
   <li> Is this a bug fix or a feature? </li>
   <li> Does it break any existing functionality?</li>
</ul>

## Goals of this PR :tada:
 <ul>
   <li> Why is the change important? </li>
   <li> What does this fix? </li>
   <li> How far has it been tested? </li>
 </ul>
 
## How Has This Been Tested :mag:

Please let us know if you have tested your PR and if we need to reproduce the issues. Also, please let us know if we need any relevant information for running the tests.

<ul>
 <li> User Interface Testing </li>
 <li> Application Testing </li>
</ul>

## Test Configuration :space_invader:

<ul>
 <li> Xcode version: </li>
 <li> Device/Simulator </li>
 <li> iOS version || MacOSX version</li>
</ul>

## Things to check on :dart:

 - [ ] My Pull Request code follows the coding standards and styles of the project 
 - [ ] I have worked on unit tests and reviewed my code to the best of my ability 
 - [ ] I have used comments to make other coders understand my code better 
 - [ ] My changes are good to go without any warnings 
 - [ ] I have added unit tests both for the happy and sad path 
 - [ ] All of my unit tests pass successfully before pushing the PR 
 - [ ] I have made sure all dependent downstream changes impacted by my PR are working

### CODE REVIEW

File: CardParts/src/Classes/Card Parts/CardPartAttributedTextView.swift [hunk 1]
```diff
@@ -96,49 +100,62 @@ public class CardPartAttributedTextView: UIView, CardPartView {
...
     }
     
     /// allows for exclusion paths to be added for text wrapping
     public var exclusionPath: [UIBezierPath]? {
         didSet {
-            let imageFrame = updateExclusionPath()
-            let exclusionPath = UIBezierPath(rect: imageFrame)
-            self.textView.textContainer.exclusionPaths = [exclusionPath]
+            DispatchQueue.main.async {
+                let imageFrame = self.updateExclusionPath()
```
Comment (croossin) [inline CardParts/src/Classes/Card Parts/CardPartAttributedTextView.swift]: Potential memory leak - weakify self

File: CardParts/src/Classes/Card Parts/CardPartAttributedTextView.swift [hunk 2]
```diff
@@ -68,13 +68,17 @@ public class CardPartAttributedTextView: UIView, CardPartView {
     /// use in cases of plain text
     public var text: String? {
         didSet {
-            textView.text = text
+            DispatchQueue.main.async {
+                self.textView.text = self.text
```
Comment (croossin) [inline CardParts/src/Classes/Card Parts/CardPartAttributedTextView.swift]: Potential memory leak - weakify self

File: CardParts/src/Classes/Card Parts/CardPartAttributedTextView.swift [hunk 3]
```diff
@@ -68,13 +68,17 @@ public class CardPartAttributedTextView: UIView, CardPartView {
...
+            }
         }
     }
     /// use in cases of links or other
     public var attributedText: NSMutableAttributedString? {
         didSet {
-            textView.attributedText = attributedText
+            DispatchQueue.main.async {
+                self.textView.attributedText = self.attributedText
```
Comment (croossin) [inline CardParts/src/Classes/Card Parts/CardPartAttributedTextView.swift]: Potential memory leak - weakify self

File: CardParts/src/Classes/Card Parts/CardPartAttributedTextView.swift [hunk 4]
```diff
@@ -96,49 +100,62 @@ public class CardPartAttributedTextView: UIView, CardPartView {
     /// set whether the text within a text field is editable
     public var isEditable: Bool {
         didSet {
-            textView.isEditable = isEditable
+            DispatchQueue.main.async {
+                self.textView.isEditable = self.isEditable
```
Comment (croossin) [inline CardParts/src/Classes/Card Parts/CardPartAttributedTextView.swift]: Potential memory leak - weakify self

File: CardParts/src/Classes/Card Parts/CardPartAttributedTextView.swift [hunk 5]
```diff
@@ -96,49 +100,62 @@ public class CardPartAttributedTextView: UIView, CardPartView {
...
+            }
         }
     }
     /// use to specify data that should automatically be passed toother applications as URLs, e.g. phone numbers
     public var dataDetectorTypes: UIDataDetectorTypes {
         didSet {
-            textView.dataDetectorTypes = dataDetectorTypes
+            DispatchQueue.main.async {
+                self.textView.dataDetectorTypes = self.dataDetectorTypes
```
Comment (croossin) [inline CardParts/src/Classes/Card Parts/CardPartAttributedTextView.swift]: Potential memory leak - weakify self

File: CardParts/src/Classes/Card Parts/CardPartAttributedTextView.swift [hunk 6]
```diff
@@ -96,49 +100,62 @@ public class CardPartAttributedTextView: UIView, CardPartView {
...
+            }
         }
     }
     /// use to specify text alignment wthin the text field
     public var textAlignment: NSTextAlignment {
         didSet {
-            textView.textAlignment = textAlignment
+            DispatchQueue.main.async {
+                self.textView.textAlignment = self.textAlignment
```
Comment (croossin) [inline CardParts/src/Classes/Card Parts/CardPartAttributedTextView.swift]: Potential memory leak - weakify self

File: CardParts/src/Classes/Card Parts/CardPartAttributedTextView.swift [hunk 7]
```diff
@@ -96,49 +100,62 @@ public class CardPartAttributedTextView: UIView, CardPartView {
...
+            }
         }
     }
     /// set attributes of linked text
     public var linkTextAttributes: [NSAttributedString.Key : Any] {
         didSet {
-            textView.linkTextAttributes = linkTextAttributes
+            DispatchQueue.main.async {
+                self.textView.linkTextAttributes = self.linkTextAttributes
```
Comment (croossin) [inline CardParts/src/Classes/Card Parts/CardPartAttributedTextView.swift]: Potential memory leak - weakify self

File: CardParts/src/Classes/Card Parts/CardPartAttributedTextView.swift [hunk 8]
```diff
@@ -96,49 +100,62 @@ public class CardPartAttributedTextView: UIView, CardPartView {
...
         }
     }
     /// set line spacing
     public var lineSpacing: CGFloat = 1.0 {
         didSet {
-            let style = NSMutableParagraphStyle()
-            style.lineSpacing = lineSpacing
-            
+            DispatchQueue.main.async {
+                let style = NSMutableParagraphStyle()
```
Comment (croossin) [inline CardParts/src/Classes/Card Parts/CardPartAttributedTextView.swift]: Potential memory leak - weakify self

File: CardParts/src/Classes/Card Parts/CardPartAttributedTextView.swift [hunk 9]
```diff
@@ -96,49 +100,62 @@ public class CardPartAttributedTextView: UIView, CardPartView {
...
         }
     }
     /// set line height
     public var lineHeightMultiple: CGFloat = 1.1 {
         didSet {
-            let style = NSMutableParagraphStyle()
-            style.lineHeightMultiple = lineHeightMultiple
+            DispatchQueue.main.async {
+                let style = NSMutableParagraphStyle()
+                style.lineHeightMultiple = self.lineHeightMultiple
```
Comment (croossin) [inline CardParts/src/Classes/Card Parts/CardPartAttributedTextView.swift]: Potential memory leak - weakify self

File: CardParts/src/Classes/Card Parts/CardPartAttributedTextView.swift (hunk 1 above)
Comment (jl-codes) [reply CardParts/src/Classes/Card Parts/CardPartAttributedTextView.swift]: I thought didSet doesn't capture anything, preventing retain cycles?

Comment (croossin) [reply CardParts/src/Classes/Card Parts/CardPartAttributedTextView.swift]: I know didSet doesn't require to weakify self but my concern was that we are now changing threads from the originally called thread... but I could very well be wrong - let me look into this a little deeper.

**Pull request merged**